src/server/res.rs

```diff
@@ -9,11 +9,27 @@
 //! Response factory functions.
 //!
 
-use headers::{ContentLength, HeaderMapExt};
+use headers::{ContentLength, Header, HeaderMapExt, HeaderValue, Location};
 use hyper::StatusCode;
 
 use crate::server::Response;
 
+/// Generate 304 NotModified response.
+pub fn moved_permanently(mut res: Response, src: &str) -> Response {
+    *res.status_mut() = StatusCode::MOVED_PERMANENTLY;
+    let location = match HeaderValue::from_str(src) {
+        Ok(location) => location,
+        Err(err) => panic!("{} cannot be formatted into a Location header: {}", src, err),
+    };
+    let location = match Location::decode(&mut std::iter::once(&location)) {
+        Ok(location) => location,
+        Err(err) => panic!("{} cannot be formatted into a Location header: {}", src, err),
+    };
```

Comment:
This can be reduced into one match expression.

```suggestion
    let location = match Location::decode(&mut HeaderValue::from_str(src).iter()) {
        Ok(location) => location,
        Err(err) => panic!("{} cannot be formatted into a Location header: {}", src, err),
    };
```

Reply:
This doesn't report errors from the from_str call, does it?

Reply:
Ah. Yep. You're right 👍🏾